libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -743,6 +758,196 @@ bool IRGeneratorForStatements::visit(Literal const& _literal)
+		let <result> := <call>(<gas>, <address>, <value>, <pos>, sub(<end>, <pos>), <pos>, <retSize>)
+		if iszero(<result>) { <forwardingRevert> }
+		<copyDynamicReturnData>
+		<allocate>
+		mstore(<freeMem>, add(<pos>, and(add(<retSize>, 0x1f), not(0x1f))))
+		<defineRetVars> <abiDecode>(<pos>, <retSize>)
+	)");
+	string pos = m_context.newYulVariable();
+	templ("pos", pos);
+	templ("end", m_context.newYulVariable());
+	templ("result", m_context.newYulVariable());
+	templ("freeMem", to_string(CompilerUtils::freeMemoryPointer));
+	templ("shl28", m_utils.shiftLeftFunction(8 * (32 - 4)));
+	templ("funId", m_context.variablePart(_functionCall.expression(), 2));
+
+	// If the function takes arbitrary parameters or is a bare call, copy dynamic length data in place.
+	// Move arguments to memory, will not update the free memory pointer (but will update the memory
+	// pointer on the stack).
+	bool encodeInPlace = funType.takesArbitraryParameters() || funType.isBareCall();
+	if (funType.kind() == FunctionType::Kind::ECRecover)
+		// This would be the only combination of padding and in-place encoding,
+		// but all parameters of ecrecover are value types anyway.
+		encodeInPlace = false;
+	bool encodeForLibraryCall = funKind == FunctionType::Kind::DelegateCall;
+	solUnimplementedAssert(!encodeInPlace, "");
+	solUnimplementedAssert(!funType.padArguments(), "");
+	templ("encodeArgs", abi.tupleEncoder(argumentTypes, funType.parameterTypes(), encodeForLibraryCall));
+	templ("argumentString", argumentString);
+
+	// Output data will replace input data, unless we have ECRecover (then, output
+	// area will be 32 bytes just before input area).
+	templ("retSize", to_string(retSize));
+	solUnimplementedAssert(funKind != FunctionType::Kind::ECRecover, "");
+
+	if (isDelegateCall)
+		solAssert(!funType.valueSet(), "Value set for delegatecall");
+	else if (useStaticCall)
+		solAssert(!funType.valueSet(), "Value set for staticcall");
+	else if (funType.valueSet())
+		templ("value", m_context.variablePart(_functionCall.expression(), 4));
+	else
+		templ("value", "0");
+
+	bool existenceChecked = false;
+	// Check the target contract exists (has code) for non-low-level calls.
+	if (funKind == FunctionType::Kind::External || funKind == FunctionType::Kind::DelegateCall)
+	{
+		templ(
+			"existenceCheck",
+			"if iszero(extcodesize(" +
+			m_context.variablePart(_functionCall.expression(), 1) +
+			")) { revert(0, 0)} }"
+		);
+		existenceChecked = true;
+	}
+	else
+		templ("existenceCheck", "");
+
+	if (funType.gasSet())
+		templ("gas", m_context.variablePart(_functionCall.expression(), 3));
+	else if (m_context.evmVersion().canOverchargeGasForCall())
+		// Send all gas (requires tangerine whistle EVM)
+		templ("gas", "gas()");
+	else
+	{
+		// send all gas except the amount needed to execute "SUB" and "CALL"
+		// @todo this retains too much gas for now, needs to be fine-tuned.
+		u256 gasNeededByCaller = eth::GasCosts::callGas(m_context.evmVersion()) + 10;
```

Comment:
Why + 10?

Reply:
As the comment says, it's the cost of `sub; call` - at least I think...

Reply:
Wie gesagt... :p
Yea, thought so